eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1192,14 +1192,19 @@ public function updateContent(APIVersionInfo $versionInfo, APIContentUpdateStruc
             null,
             $versionInfo->versionNo
         );
+
         if (!$content->versionInfo->isDraft()) {
             throw new BadStateException(
                 '$versionInfo',
                 'Version is not a draft and can not be updated'
             );
         }
 
-        if (!$this->repository->canUser('content', 'edit', $content)) {
+        if ($this->isNewTranslation($versionInfo, $contentUpdateStruct)) {
+            if (!$this->repository->canUser('content', 'translate', $contentUpdateStruct)) {
+                throw new UnauthorizedException('content', 'translate', array('contentId' => $content->id));
```

Comment:
What will happen if I have an editor created in v2.3 or in older version and he doesn't have `content/translate` policy? Was he able to add translation then?

Reply:
no

Maybe we should check translate || edit in this case, not just for bc, but to signify edit permission also implies translation permission (??)